### PR TITLE
[improve](backup) Reduce backup/restore job log size by compress

### PR DESCRIPTION
**Pull request by w41ter**

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

**Comment by w41ter** [comment]
run buildall

**Comment by w41ter** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/persist/gson/GsonUtils.java [hunk 1]
```diff
@@ -978,4 +986,26 @@ public T read(JsonReader reader) throws IOException {
...
     }
 
+    public static void toJsonCompressed(DataOutput out, Object src) throws IOException {
+        ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+        try (GZIPOutputStream gzipStream = new GZIPOutputStream(byteStream)) {
+            try (OutputStreamWriter writer = new OutputStreamWriter(gzipStream)) {
+                GsonUtils.GSON.toJson(src, writer);
+            }
+        }
+        Text text = new Text(byteStream.toByteArray());
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/persist/gson/GsonUtils.java:996]: image
Is it correct to use `Text`?

**Review by swjtu-zhanglei — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/persist/gson/GsonUtils.java (hunk 1 above)
Comment (w41ter) [reply fe/fe-core/src/main/java/org/apache/doris/persist/gson/GsonUtils.java:996]: `Text` also supports writing bytes directly without encoding.

**Pull request merged**